tests/profiles-enabled/test_shipments.py

```diff
@@ -1652,16 +1664,44 @@ def test_shipment_set_shadow(self):
             assert response.status_code == status.HTTP_400_BAD_REQUEST
             assert mocked.call_count == mocked_call_count
 
+            # Removing device should fail if there is no delivery_act on the shipment
+            response = self.set_device_id(shipment['id'], None, None)
+            assert response.status_code == status.HTTP_400_BAD_REQUEST
+            assert mocked.call_count == mocked_call_count
+
+            # Removing device should fail if delivery_act is in the future
+            shipment_obj.refresh_from_db()
+            shipment_obj.delivery_act = datetime.now() + timedelta(days=1)
+            shipment_obj.save()
+            mocked_call_count += 2  # Shipment_obj.save() increases the call count
+
+            response = self.set_device_id(shipment['id'], None, None)
+            assert response.status_code == status.HTTP_400_BAD_REQUEST
+            assert mocked.call_count == mocked_call_count
+
             # Devices can be reused after deliveries are complete
             shipment_obj.refresh_from_db()
             shipment_obj.delivery_act = datetime.now()
             shipment_obj.save()
 
             response = self.create_shipment()
             assert response.status_code == status.HTTP_202_ACCEPTED
-            mocked_call_count += 3
```

Comment:
You changed this test case, which was originally testing that a device should be able to be 'reassigned' to a new shipment if its previous shipment is delivered (and the device is still attached to the old shipment).

Reply:
This is still being tested. I did not remove the device from the shipment from that one, it gets removed only in line 1685, which is one of the tests from above that I commented out